File: java/src/main/java/factory/product/SuperBus.java
package factory.product;

import factory.product.part.Engine;
import factory.product.part.Tire;

public class SuperBus extends Vehicle {
    public static final String NAME = "Super Bus";

    public SuperBus() {
        name = NAME;
        engine = Engine.HYBRID;
        tire = Tire.NORMAL;
    }
}

File: src/main/kotlin/factory/product/SuperBus.kt
package factory.product

import factory.product.part.Engine
import factory.product.part.Tire

class SuperBus : Vehicle() {
    override var name = NAME
    override var engine: Engine = Engine.HYBRID
    override var tire: Tire = Tire.NORMAL

    companion object {
        const val NAME: String = "Super Bus"
    }
}